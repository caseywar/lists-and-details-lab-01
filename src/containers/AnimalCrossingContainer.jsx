import React, { Component } from 'react'
import VillagerList from '../components/characters/VillagerList';
import { fetchVillagers } from '../services/AnimalCrossingApi';

export default class AnimalCrossingContainer extends Component {
    state = {
        loading: true,
        villagers: [],
    };

    async componentDidMount() {
        const villagers = await fetchVillagers();
        this.setState({
            villagers,
            loading: false,
        });
    };

    render() {
        const { loading, villagers } = this.state;

        if (loading) return <h1>Loading...</h1>;

        return <VillagerList villagers={villagers} />
    }
}
