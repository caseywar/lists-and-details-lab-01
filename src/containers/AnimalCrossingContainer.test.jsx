import React from 'react';
import { render, screen } from '@testing-library/react';
import AnimalCrossingContainer from './AnimalCrossingContainer';
import { MemoryRouter } from 'react-router';


describe('Animal Crossings Container', () => {
    it('renders a list of characters on the web page', async () => {
        render(   
        <MemoryRouter>
            <AnimalCrossingContainer />
          </MemoryRouter>);

        screen.getByText('Loading...');

        const ul = await screen.findAllByRole( 'list', {name: 'villagers'});
        expect(ul).toMatchSnapshot();
    });
});
