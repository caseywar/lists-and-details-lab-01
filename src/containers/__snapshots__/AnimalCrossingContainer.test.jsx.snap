// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Crossings Container renders a list of characters on the web page 1`] = `
Array [
  <ul
    aria-label="villagers"
    class="list"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e460"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Admiral"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
          />
          <p>
            Admiral
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e461"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Agent S"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
          />
          <p>
            Agent S
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e462"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Agnes"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
          />
          <p>
            Agnes
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e463"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Al"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
          />
          <p>
            Al
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e464"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Alfonso"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
          />
          <p>
            Alfonso
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e465"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Alice"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
          />
          <p>
            Alice
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46c"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Ankha"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
          />
          <p>
            Ankha
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46a"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Angus"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
          />
          <p>
            Angus
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e466"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Alli"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
          />
          <p>
            Alli
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46b"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Anicotti"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
          />
          <p>
            Anicotti
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e467"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Amelia"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
          />
          <p>
            Amelia
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46d"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Annalisa"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
          />
          <p>
            Annalisa
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46f"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Antonio"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
          />
          <p>
            Antonio
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46e"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Annalise"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
          />
          <p>
            Annalise
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e468"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Anabelle"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
          />
          <p>
            Anabelle
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e470"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Apollo"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
          />
          <p>
            Apollo
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e469"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Anchovy"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
          />
          <p>
            Anchovy
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e471"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Apple"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
          />
          <p>
            Apple
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e472"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Astrid"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
          />
          <p>
            Astrid
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e473"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Audie"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
          />
          <p>
            Audie
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e474"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Aurora"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
          />
          <p>
            Aurora
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e476"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Avery"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
          />
          <p>
            Avery
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e475"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Ava"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
          />
          <p>
            Ava
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e47b"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Barold"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
          />
          <p>
            Barold
          </p>
        </div>
      </li>
    </a>
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e47a"
    >
      <li
        class="character"
      >
        <div>
          <img
            alt="Bangle"
            src="https://vignette.wikia.nocookie.net/animalcrossing/images/6/68/Bangle_NH.png/revision/latest?cb=20200803120044"
          />
          <p>
            Bangle
          </p>
        </div>
      </li>
    </a>
  </ul>,
]
`;
